proof/refine/ARM/Syscall_R.thy

```diff
@@ -920,22 +928,39 @@ lemma threadSet_not_curthread_ct_domain:
   apply clarsimp
   done
 
+lemma tcbSchedDequeue_not_in_queue:
+  "\<lbrace>valid_queues\<rbrace>
+   tcbSchedDequeue t
+   \<lbrace>\<lambda>rv s. t \<notin> set (ksReadyQueues s p)\<rbrace>"
```

Comment:
Nice! Could you please replace `tcbSchedDequeue_nonq` with this one?

Reply:
Ah, I thought such a lemma already existed!!